resource_type.go

```diff
@@ -76,12 +82,34 @@ func (t ResourceType) validate(raw []byte) (ResourceAttributes, *errors.ScimErro
 			return ResourceAttributes{}, scimErr
 		}
 
-		attributes[extension.Schema.ID] = extensionAttributes
+		schemaAttributes[extension.Schema.ID] = extensionAttributes
+	}
+
+	attributes := make(map[string]interface{}, len(commonAttributes)+len(schemaAttributes))
+
+	for k, v := range schemaAttributes {
+		attributes[k] = v
+	}
+
+	for k, v := range commonAttributes {
+		attributes[k] = v
 	}
```

Comment:
If I am not mistaken the `k` common attributes will overwrite the first `k` schema attributes.

Reply:
Only if they share the same attribute names I think

Reply:
Oh, yeah, my bad.